Bugfix #7319 - Fixed incorrect saving for HabitTranslation for ua fields.

ChernenkoVitaliy opened the pull request:
# GreenCity PR
## Issue Link 📋
https://github.com/ita-social-projects/GreenCity/issues/7319

## Added

- Added _getHabitTranslationDtoEnAndUa_ method in ModelUtils; 
- Added `@NotBlank` annotation in HabitTranslationDto for description, languageCode and name fields; 
- Added `@Valid `annotation for validation HabitTranslationDto into CustomHabitDtoRequest;
- Added _convertUa_() and _mapAllToList_() methods into HabitTranslationMapper; 
- Added necessary tests in HabitTranslationMapperTests;

## Changed

- Changed functionality  of method saveHabitTranslationListsToHabitTranslationRepo in HabitServiceImpl;
- Modified tests according to change in method _saveHabitTranslationListsToHabitTranslationRepo_ in HabitServiceImplTests;

## Summary Of Changes 🔥
Improvement according to task https://github.com/ita-social-projects/GreenCity/issues/7319;
Fixed main functionality of saving HabitTranslation in HabitServiceImpl;

ChernenkoVitaliy committed:
Fixed functionality of saving HabitTranslation in saveHabitTranslationListsToHabitTranslationRepo method in HabitServiceImpl;
Fixed necessary tests in HabitServiceImplTest;
Added getHabitTranslationDtoEnAndUa method in ModelUtils;
Added @NotBlank annotation in HabitTranslationDto for description, languageCode and name fields;
Added @Valid annotation for validation HabitTranslationDto into CustomHabitDtoRequest;
Added convertUa() and mapAllToList() methods into HabitTranslationMapper;
Added necessary nests in HabitTranslationMapperTests;

ChernenkoVitaliy committed:
Fixed sonar issues in HabitTranslationMapperTests;

RomaMocherniuk commented on service/src/main/java/greencity/mapping/HabitTranslationMapper.java:
```diff
@@ -29,4 +60,22 @@ protected HabitTranslation convert(HabitTranslationDto habitTranslationDto) {
+     * @author Chernenko Vitaliy
+     */
+    public List<HabitTranslation> mapAllToList(List<HabitTranslationDto> dtoList, String language) {
+        if (language.equals("ua")) {
```
please move `ua` to the constant and compare like this to prevent NPE
`"ua".equals(language)`

ChernenkoVitaliy replied:
Good point. I'll remember it.
Fixed.

RomaMocherniuk commented on service/src/main/java/greencity/service/HabitServiceImpl.java:
```diff
@@ -491,21 +491,26 @@ private void updateHabitTranslationsForCustomHabit(CustomHabitDtoRequest habitDt
     private void saveHabitTranslationListsToHabitTranslationRepo(CustomHabitDtoRequest habitDto, Habit habit) {
-        List<HabitTranslation> habitTranslationListForUa = mapHabitTranslationFromAddCustomHabitDtoRequest(habitDto);
+        final String UA = "ua";
+        final String EN = "en";
```
please move to the constants

ChernenkoVitaliy replied:
Thanks. Done

RomaMocherniuk commented on service/src/main/java/greencity/mapping/HabitTranslationMapper.java:
```diff
@@ -18,6 +18,37 @@ protected HabitTranslation convert(HabitTranslationDto habitTranslationDto) {
+     */
+    public HabitTranslation convertUa(HabitTranslationDto habitTranslationDto) {
+        HabitTranslation habitTranslation = new HabitTranslation();
+        if (habitTranslationDto.getNameUa() == null) {
```
that logic can be simplified using  [defaultIfNull](https://commons.apache.org/proper/commons-lang/javadocs/api-3.1/src-html/org/apache/commons/lang3/ObjectUtils.html#line.93)

ChernenkoVitaliy replied:
Very useful thing. I think I'll use it in the future.
Thanks. Fixed.

RomaMocherniuk commented:
Please increase test coverage to at least 90%
<img width="485" alt="image" src="https://github.com/user-attachments/assets/16d5ae78-509c-4ad9-b90d-0dc9ac9ac208">

ChernenkoVitaliy committed:
Moved strings for language codes in AppConstant;
Replced if/else to ObjectUtils.defaultNull;
Added test for HabitTranslationMapperTests to improve coverage;

ChernenkoVitaliy committed:
Formatted code in HabitTranslationMapperTests with formatter-maven-plagin;

ChernenkoVitaliy commented:
> Please increase test coverage to at least 90% <img alt="image" width="485" src="https://private-user-images.githubusercontent.com/15122465/394642650-16d5ae78-509c-4ad9-b90d-0dc9ac9ac208.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzM5MjY0MzcsIm5iZiI6MTczMzkyNjEzNywicGF0aCI6Ii8xNTEyMjQ2NS8zOTQ2NDI2NTAtMTZkNWFlNzgtNTA5Yy00YWQ5LWI5MGQtMGRjOWFjOWFjMjA4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMTElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjExVDE0MDg1N1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTExMWM3M2Y5MWZmZTFkZThiZGUxNDcwN2I0ZWQ2NTY0MmNjZThlYjNkYTdhMzMzMzUzNzYxNDk4MDVlM2Y5YWUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.5hPr7R5NP-ojFcrDkpYVH_yDYJWsNadXICw9u4thR18">

Thanks. Improved.
For now 100% covered.
